servers/shared-snowplow-consumer/src/snowplow/user/userEventHandler.ts

```diff
@@ -44,118 +34,107 @@ export class UserEventHandler extends EventHandler {
+      object: 'account',
     };
   }
 
   /**
    * @private to build event context for ACCOUNT_DELETE event.
    */
   private static generateDeleteEventAccountContext(
-    data: UserEventPayloadSnowplow,
-  ): AccountContext {
-    return {
-      schema: userEventsSchema.account,
-      data: {
-        object_version: 'new',
-        user_id: parseInt(data.user.id),
-      },
-    };
+    data: UserEventBridgePaylod['detail'],
+  ): SelfDescribingJson {
+    return createAccount({
+      object_version: 'new',
+      user_id: parseInt(data.userId),
+    }) as unknown as SelfDescribingJson;
   }
 
   private static generateAccountContext(
-    data: UserEventPayloadSnowplow,
-  ): AccountContext {
-    return {
-      schema: userEventsSchema.account,
-      data: {
-        object_version: 'new',
-        user_id: parseInt(data.user.id),
-        emails: [data.user.email],
-      },
-    };
+    data: UserEventBridgePaylod['detail'],
+  ): SelfDescribingJson {
+    return createAccount({
+      object_version: 'new',
+      user_id: parseInt(data.userId),
+      emails: [data.email],
+    }) as unknown as SelfDescribingJson;
   }
 
   private static generateEventContext(
-    data: UserEventPayloadSnowplow,
+    data: UserEventBridgePaylod,
   ): SelfDescribingJson[] {
     const context = [
-      UserEventHandler.generateUserContext(data),
-      UserEventHandler.generateApiUserContext(data),
+      UserEventHandler.generateUserContext(data.detail),
+      UserEventHandler.generateApiUserContext(data.detail),
     ];
 
-    data.eventType == EventType.ACCOUNT_DELETE
-      ? context.push(UserEventHandler.generateDeleteEventAccountContext(data))
-      : context.push(UserEventHandler.generateAccountContext(data));
+    data['detail-type'] == 'account-deletion'
+      ? context.push(
+          UserEventHandler.generateDeleteEventAccountContext(data.detail),
+        )
+      : context.push(UserEventHandler.generateAccountContext(data.detail));
     return context;
   }
 
   private static generateUserContext(
-    data: UserEventPayloadSnowplow,
-  ): UserContext {
+    data: UserEventBridgePaylod['detail'],
+  ): SelfDescribingJson {
     const userDataWithoutGuid = {
-      email: data.user.email,
-      hashed_guid: data.user.hashedGuid,
-      user_id: parseInt(data.user.id),
-      hashed_user_id: data.user.hashedId,
+      email: data.email,
+      hashed_guid: data.hashedGuid,
+      user_id: parseInt(data.userId),
+      hashed_user_id: data.hashedId,
     };
 
-    if (data.user.guid) {
-      return {
-        schema: userEventsSchema.user,
-        data: { ...userDataWithoutGuid, guid: data.user.guid },
-      };
+    if (data.guid) {
+      return createUser({
+        ...userDataWithoutGuid,
+        guid: data.guid,
+      }) as unknown as SelfDescribingJson;
     }
 
-    return {
-      schema: userEventsSchema.user,
-      data: { ...userDataWithoutGuid },
-    };
+    return createUser({
```

Comment:
Is the behavior different if `guid` is undefined vs. omitted completely? if not, this whole function could be removed